woocommerce/woocommerce
# Display shipping calculator link in the Cart block when fallback shipping options available

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #42280.

This PR fixes the scenarios to display the "change address" link in the cart block.

- When `Default customer location` is set to  `No default location` with Multiple shipping zones and fallback shipping zone: Display the shipping calculator with `Calculate shipping for your location` generic text. 
- Shop with no shipping zones defined: Hide the "change address" link from the editor.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Disable local pickup from wp-admin/admin.php?page=wc-settings&tab=shipping&section=pickup_location.  
2. Delete all shipping zones. 
3. Go to shipping settings at wp-admin/admin.php?page=wc-settings&tab=shipping&section=options.
4. Disable the "Hide shipping costs until an address is entered" option. 
5. Activate the "Enable the shipping calculator on the cart page" option.
![image](https://github.com/woocommerce/woocommerce/assets/11503784/6952c0d6-e7bb-4863-8ecb-1a9c64dcc3a7)

6. Verify "Default customer location" is "no location by default". 
7. Navigate to the Cart block editor. 
8. Confirm the no shipping sections available in the Editor.

<img width="1319" alt="image" src="https://github.com/woocommerce/woocommerce/assets/11503784/ab42d4ac-2c06-4c97-9bf6-e0582c2a864a">

9. Add one shipping zone for the US with a free shipping option. 
10. Add a flat rate shipping option for all shipping zones. 

<img width="1550" alt="image" src="https://github.com/woocommerce/woocommerce/assets/11503784/3c2c95a9-d5d0-4109-89ff-4bd43a0e27f8">

11. Visit the site in a private window as a guest user. 
12. Add the product to the cart and visit the Cart page. 
13. Confirm it displays the flat rate shipping method with the `Calculate shipping for your location` link. 

<img width="1366" alt="image" src="https://github.com/woocommerce/woocommerce/assets/11503784/55c7b135-3c98-4379-94c2-cbcd0b030281">

14. Remove the flat rate shipping option for all shipping zones. (that was added in Step 8). 
15. Add flat rate shipping option for `Rest of the world`. 
![image](https://github.com/woocommerce/woocommerce/assets/11503784/e5f0f514-3ffc-4422-be46-71beb8abbcd0)

16.  Visit the site in a private window as a guest user. 
17. Add the product to the cart and visit the Cart page. 
18. Confirm it displays the flat rate shipping method with the `Calculate shipping for your location` link. 
<img width="1366" alt="image" src="https://github.com/woocommerce/woocommerce/assets/11503784/55c7b135-3c98-4379-94c2-cbcd0b030281">

19. Go to shipping settings at wp-admin/admin.php?page=wc-settings&tab=shipping&section=options
20. Enable the "Hide shipping costs until an address is entered" option. 
21. Visit the site in a private window as a guest user. 
22. Add the product to the cart, visit the Cart page, and confirm it didn't display the shipping methods. 
<img width="1311" alt="image" src="https://github.com/woocommerce/woocommerce/assets/11503784/744e1d2f-88d3-4b3d-9efc-e230fbc828e2">




### Changelog entry




<details>

#### Significance




#### Type




#### Message 
Hide the shipping calculator link on the Cart editor with no default address and zone. Display the shipping calculator link on the front end for multiple zones with a fallback.

#### Comment 

</details>
